### DISCUSSION TITLE
Ordering by multiple columns

### DISCUSSION THREAD

**Discussion by Cannonb4ll** [filamentphp/filament#2626] [Ideas]
As of now, you can order in a table by using the method `->defaultSort('created_at', 'desc')`, this is great, but it would be even more awesome if you can sort by multiple columns.

Something in the lines of:

```php
->defaultSort([
    'created_at' => 'desc',
    'last_login_at' => 'desc'
]);
```

**Comment by danharrin** [comment]
This is possible in v3:

```php
->defaultSort(fn ($query) => $query->orderBy('created_at', 'desc')->orderBy('last_login_at', 'desc'))
```

**Comment by MaxAIPlayground** [comment]
Is it possible to apply a custom orderByRaw() when a (default) filter is selected/active?

This is what I have:

```php
    public static function getEloquentQuery(): Builder
    {
        return parent::getEloquentQuery()
            ->withoutGlobalScopes([
                'approved',
                SoftDeletingScope::class,
            ])
            ->orderByRaw(
                 "CASE WHEN approved_at IS NULL THEN id ELSE upvotes_count END DESC"
            );
    }
```
    
I also use a default filter, but when I unselect the default filter, I'd like to apply orderBy('id') instead of the orderByRaw().
Or put in another way: My default filter should apply the custom orderByRaw() and when this default filter is unselected (that means no filter is currently active) I'd like to apply an orderBy('id') on the records.

This is my default filter:
```php
                Filter::make('not_approved_not_answered')
                    ->label(__('filament::resources/questions.not_approved_not_answered'))
                    ->query(function (Builder $query) {
                        $query->whereNull('approved_at')
                            ->orWhereNull('answered_at');
                           // Applying the orderByRaw() here will not have any effect...
                    })
                    ->default(),
```
